### PR TITLE
add channel key support

### LINKED ISSUE #6: support joining channel with key (+k)
hey,
i actually need this feature at irc_bot for usage of this bot in flexget https://github.com/Flexget/Flexget.

Correct me if i'm wrong but i cant join a channel protected with a key (mode +k), right?

Would be great if this will be implemented, thank you

### PR DESCRIPTION
**Pull request by ekilah**
Fixes #6 

Adds support for joining channels that have a key / password / `mode +k`. You provide the key next to the channel name:

instead of `'#channel-name'`
you do `'#channel-name YourKeyHere'`

### CODE REVIEW

**Comment by cvium** [comment]
Can a channel key contain spaces? Since you only want to split it into two, you should call split(string, 1) instead.

I'm also not sure I like the whitespace as delimiter. Comma is not allowed in a channel name either, so it could be used as well.

Thoughts?

**Comment by ekilah** [comment]
I'm very new to IRC, and I didn't go looking for rules about channel names or keys re: whitespace. Good call/question.

Comma as a delimiter seems slightly more complicated because you might want to allow for whitespace around it, requiring a regex etc.

I can make the split change to theoretically allow whitespace in keys, sure!

File: irc_bot/simple_irc_bot.py [hunk 1]
```diff
@@ -444,7 +463,7 @@ def identify_with_nickserv(self):
 
             # Identify with NickServ
             log.info('Identifying with NickServ as %s', self.nickname)
-            self.send_privmsg('NickServ', 'IDENTIFY %s' % self.nickserv_password)
+            self.send_privmsg('NickServ', 'IDENTIFY %s %s' % (self.nickname, self.nickserv_password))
```
Comment (ekilah) [inline irc_bot/simple_irc_bot.py]: I did this while debugging. I can undo it, it shouldn't be necessary since `self.nick` is called earlier. Doesn't hurt

**Review by cvium — APPROVED** [review]

**Comment by cvium** [comment]
Thanks. I have pushed a new release.

**Comment by ekilah** [comment]
Thanks for the quick responses!

**Pull request merged**